Print formatting

ambaker1 opened the pull request:
This is to make the output of "printModel" more readable.
I added header lines, made the headers more consistent, and fixed some formatting issues.

ambaker1 commented:
Old print (using Truss.tcl as an example):
  
    Current Domain Information
            Current Time: 1
    tCommitted Time: 1
    NODE DATA: NumNodes: 4
    
    numComponents: 4
     Node: 1
            Coordinates  : 0 0
            Disps: 0 0
            Velocities   : 0 0
             unbalanced Load: 0 0
            ID : -1 -1
    
    
     Node: 2
            Coordinates  : 144 0
            Disps: 0 0
            Velocities   : 0 0
             unbalanced Load: 0 0
            ID : -1 -1
    
    
     Node: 3
            Coordinates  : 168 0
            Disps: 0 0
            Velocities   : 0 0
             unbalanced Load: 0 0
            ID : -1 -1
    
    
     Node: 4
            Coordinates  : 72 96
            Disps: 0.530093 -0.177894
            Velocities   : 0 0
             unbalanced Load: 100 -50
            ID : 0 1
    
    ELEMENT DATA: NumEle: 3
    
    numComponents: 3Element: 1 type: Truss  iNode: 1 jNode: 4 Area: 10 Mass/Length: 0 cMass: 0
             strain: 0.00146451 axial load: 43.9352
             unbalanced load: -26.3611 -35.1482 26.3611 35.1482
             Material:
    Element: 2 type: Truss  iNode: 2 jNode: 4 Area: 5 Mass/Length: 0 cMass: 0
             strain: -0.00383642 axial load: -57.5463
             unbalanced load: -34.5278 46.0371 34.5278 -46.0371
             Material:
    Element: 3 type: Truss  iNode: 3 jNode: 4 Area: 5 Mass/Length: 0 cMass: 0
             strain: -0.00368743 axial load: -55.3114
             unbalanced load: -39.1111 39.1111 39.1111 -39.1111
             Material:
    
    SP_Constraints: numConstraints: 6
    
    numComponents: 6SP_Constraint: 0         Node: 1 DOF: 1 ref value: 0 current value: 0
    SP_Constraint: 1         Node: 1 DOF: 2 ref value: 0 current value: 0
    SP_Constraint: 2         Node: 2 DOF: 1 ref value: 0 current value: 0
    SP_Constraint: 3         Node: 2 DOF: 2 ref value: 0 current value: 0
    SP_Constraint: 4         Node: 3 DOF: 1 ref value: 0 current value: 0
    SP_Constraint: 5         Node: 3 DOF: 2 ref value: 0 current value: 0
    
    Pressure_Constraints: numConstraints: 0
    
    numComponents: 0
    MP_Constraints: numConstraints: 0
    
    numComponents: 0
    LOAD PATTERNS: numPatterns: 1
    
    
    numComponents: 1Load Pattern: 1
      Scale Factor: 1
    Linear Series: constant factor: 1
      Nodal Loads:
    
    numComponents: 1Nodal Load: 4 load : 100 -50
    
      Elemental Loads:
    
    numComponents: 0
      Single Point Constraints:
    
    numComponents: 0
    PARAMETERS: numParameters: 0
    
    
    numComponents: 0

New print:

    CURRENT DOMAIN INFORMATION:
    -------------------------------------------------------------------------------
    Current Time: 1
    Committed Time: 1
    
    NODES:
    -------------------------------------------------------------------------------
    numComponents: 4
    
     Node: 1
            Coordinates  : 0 0
            Disps: 0 0
            Velocities   : 0 0
             unbalanced Load: 0 0
            ID : -1 -1
    
    
     Node: 2
            Coordinates  : 144 0
            Disps: 0 0
            Velocities   : 0 0
             unbalanced Load: 0 0
            ID : -1 -1
    
    
     Node: 3
            Coordinates  : 168 0
            Disps: 0 0
            Velocities   : 0 0
             unbalanced Load: 0 0
            ID : -1 -1
    
    
     Node: 4
            Coordinates  : 72 96
            Disps: 0.530093 -0.177894
            Velocities   : 0 0
             unbalanced Load: 100 -50
            ID : 0 1
    
    
    ELEMENTS:
    -------------------------------------------------------------------------------
    numComponents: 3
    Element: 1 type: Truss  iNode: 1 jNode: 4 Area: 10 Mass/Length: 0 cMass: 0
             strain: 0.00146451 axial load: 43.9352
             unbalanced load: -26.3611 -35.1482 26.3611 35.1482
             Material:
    Element: 2 type: Truss  iNode: 2 jNode: 4 Area: 5 Mass/Length: 0 cMass: 0
             strain: -0.00383642 axial load: -57.5463
             unbalanced load: -34.5278 46.0371 34.5278 -46.0371
             Material:
    Element: 3 type: Truss  iNode: 3 jNode: 4 Area: 5 Mass/Length: 0 cMass: 0
             strain: -0.00368743 axial load: -55.3114
             unbalanced load: -39.1111 39.1111 39.1111 -39.1111
             Material:
    
    SP CONSTRAINTS:
    -------------------------------------------------------------------------------
    numComponents: 6
    SP_Constraint: 0         Node: 1 DOF: 1 ref value: 0 current value: 0
    SP_Constraint: 1         Node: 1 DOF: 2 ref value: 0 current value: 0
    SP_Constraint: 2         Node: 2 DOF: 1 ref value: 0 current value: 0
    SP_Constraint: 3         Node: 2 DOF: 2 ref value: 0 current value: 0
    SP_Constraint: 4         Node: 3 DOF: 1 ref value: 0 current value: 0
    SP_Constraint: 5         Node: 3 DOF: 2 ref value: 0 current value: 0
    
    PRESSURE CONSTRAINTS:
    -------------------------------------------------------------------------------
    numComponents: 0
    
    MP CONSTRAINTS:
    -------------------------------------------------------------------------------
    numComponents: 0
    
    LOAD PATTERNS:
    -------------------------------------------------------------------------------
    numComponents: 1
    Load Pattern: 1
      Scale Factor: 1
    Linear Series: constant factor: 1
      Nodal Loads:
    numComponents: 1
    Nodal Load: 4 load : 100 -50
    
      Elemental Loads:
    numComponents: 0
    
      Single Point Constraints:
    numComponents: 0
    
    PARAMETERS:
    -------------------------------------------------------------------------------
    numComponents: 0

silviamazzoni commented:
the extra ------- and placing them after the declaration of the output will mess up current parsing scripts.
if you are going to add separators, put them between items, please

ambaker1 commented:
@silviamazzoni yes, you make a good point - this would mess up any parsing script that parses the output of print, so this tradeoff must be considered. 
However, even if I moved the separators above the headers, this commit would still likely break any parsing script for the print output, since I changed the headers to all be consistent in format.
The intention of this PR is simply to make the text output more readable. For parsing model information, it is probably easier to use the -JSON print option. 
Anyone else have any thoughts on this?

silviamazzoni commented:
please do NOT make changes to OpenSees that are not an improvement!

zhuminjie commented:
You can add an option such as "-beautify" to the print command to add the extra decoration.

ambaker1 commented:
@silviamazzoni I can see you are not a fan of this change. Noted.
@zhuminjie Good idea! Alternatively, there could be a -legacy option for the old format, so that old parsers can still work fine, what do you think?

zhuminjie commented:
If it's me, I will leave the current outputs as default, and new outputs with an option. Because if you require -legacy, you still break other's code. Changes should be backward compatible.

ambaker1 commented:
@zhuminjie good point. I'll add a -pretty option then. 
I'll still fix some formatting issues with the original print that render it very difficult to parse or read, and I'll post the updated print outputs here.

mhscott commented:
Yeah, this change is unnecessary, so add an option for the new format

ambaker1 commented:
Since it is clear that major changes to the print format are not desired, I am instead just proposing a couple small formatting fixes. Here's the fixed print output:

    Current Domain Information
            Current Time: 1
            Committed Time: 1
    NODE DATA: NumNodes: 4
    
    numComponents: 4
    
     Node: 1
            Coordinates  : 0 0
            Disps: 0 0
            Velocities   : 0 0
             unbalanced Load: 0 0
            ID : -1 -1
    
    
     Node: 2
            Coordinates  : 144 0
            Disps: 0 0
            Velocities   : 0 0
             unbalanced Load: 0 0
            ID : -1 -1
    
    
     Node: 3
            Coordinates  : 168 0
            Disps: 0 0
            Velocities   : 0 0
             unbalanced Load: 0 0
            ID : -1 -1
    
    
     Node: 4
            Coordinates  : 72 96
            Disps: 0.530093 -0.177894
            Velocities   : 0 0
             unbalanced Load: 100 -50
            ID : 0 1
    
    ELEMENT DATA: NumEle: 3
    
    numComponents: 3
    Element: 1 type: Truss  iNode: 1 jNode: 4 Area: 10 Mass/Length: 0 cMass: 0
             strain: 0.00146451 axial load: 43.9352
             unbalanced load: -26.3611 -35.1482 26.3611 35.1482
             Material:
    Element: 2 type: Truss  iNode: 2 jNode: 4 Area: 5 Mass/Length: 0 cMass: 0
             strain: -0.00383642 axial load: -57.5463
             unbalanced load: -34.5278 46.0371 34.5278 -46.0371
             Material:
    Element: 3 type: Truss  iNode: 3 jNode: 4 Area: 5 Mass/Length: 0 cMass: 0
             strain: -0.00368743 axial load: -55.3114
             unbalanced load: -39.1111 39.1111 39.1111 -39.1111
             Material:
    
    SP_Constraints: numConstraints: 6
    
    numComponents: 6
    SP_Constraint: 0         Node: 1 DOF: 1 ref value: 0 current value: 0
    SP_Constraint: 1         Node: 1 DOF: 2 ref value: 0 current value: 0
    SP_Constraint: 2         Node: 2 DOF: 1 ref value: 0 current value: 0
    SP_Constraint: 3         Node: 2 DOF: 2 ref value: 0 current value: 0
    SP_Constraint: 4         Node: 3 DOF: 1 ref value: 0 current value: 0
    SP_Constraint: 5         Node: 3 DOF: 2 ref value: 0 current value: 0
    
    Pressure_Constraints: numConstraints: 0
    
    numComponents: 0
    
    MP_Constraints: numConstraints: 0
    
    numComponents: 0
    
    LOAD PATTERNS: numPatterns: 1
    
    
    numComponents: 1
    Load Pattern: 1
      Scale Factor: 1
    Linear Series: constant factor: 1
      Nodal Loads:
    
    numComponents: 1
    Nodal Load: 4 load : 100 -50
    
      Elemental Loads:
    
    numComponents: 0
    
      Single Point Constraints:
    
    numComponents: 0
    
    PARAMETERS: numParameters: 0
    
    
    numComponents: 0

The only changes are to the third line "Committed Time", and to add a newline after every "numComponents". In the original print output, this would often be printed on the same line as other data with no separator, making it very difficult to read and parse.
